### DISCUSSION TITLE
Diode instance parameters: AREA vs M for scaling

### DISCUSSION THREAD

**Discussion by ngwood** [Xyce/Xyce#51] [Q&A] [answered]
The `AREA` and `M` instance parameters have always served the same purpose in my mind. I've noticed that the Xyce open diode models use both, but they do not behave exactly the same though. This simple netlist is enough to demonstrate this.

```
Open diode model AREA vs M example
.model default d
v1 1 0 dc 0
d1 1 0 default
d2 1 0 default area=2
d3 1 0 default m=2
.dc v1 0 1 0.01
.print dc v(1) i(d1) i(d2) i(d3)
.end
```
![diode_area_vs_m cir prn](https://user-images.githubusercontent.com/72522985/147692096-f96ff0b6-bd46-47dd-86de-b09ff74c8022.png)

Is anyone aware of the difference? Is there even supposed to be a difference?

Is there a reason why the open JFET model has an `AREA` parameter, but not an `M` parameter? Is the `AREA` parameter supposed to behave as the `M` parameter for the JFET?

I can see that the `M` parameter was added to the open diode models only very recently; see commit `1ccccd5a181ffd7159f0331ad38a43a3c51d3b10`. The commit message implies that the `AREA` parameter was not acting as a multiplicity factor or perhaps just that the name was not suitable in some contexts. This commit was to address GitLab issue 264 (Add multiplicity to levels 1 and 2 diodes) so this was clearly a conscious decision; the description of this issue makes no mention of the existing `AREA` parameter though, which seems strange to me.

Would one of the Xyce Team be able to add some context to this perhaps?

**Comment by tvrusso** [comment, marked as answer]
The SPICE3F5 diode and JFET only have AREA instance parameters, and these serve the same purpose as "M" as you note.  Xyce's JFET and diode models are basically taken from SPICE3F5 and until recently only had these ways of scaling the device.

The diode also has an "M" MODEL parameter that has nothing to do with multiplicity.

Addition of an "M" instance parameter for multiplicity was _strictly_ an addition needed to improve Xyce's compatibility with simulators other than SPICE3F5, many of which provide multiplicity for *every* device using the "M" parameter.

We are still a long way from universal availability of multiplicity parameters, but it is our intent for every device eventually to support it.

The JFET device in Xyce has gotten very little attention since it was first implemented back in 2003, as it is not used much internally.  It will get an "M" instance parameter for multiplicity eventually.

**Reply by tvrusso** [reply]
Now, why "area=2" and "m=2" aren't producing identical results is curious, but not really surprising given the tiny size of the difference and how they're actually applied.

"M" is applied by computing all the currents and capacitor charges in the device as if there were only one, and then multiplying by "M" when adding those contributions to the DAE.  

"Area" is applied simply by scaling the saturation current, the parasitic resistor conductance, and capacitance, then computing the device  contributions using these scaled quantities.  While ideally these should produce identical results in infinite precision, I would expect some small numerical differences to creep in as a result of the difference in the way they're evaluated.

**Reply by tvrusso** [reply]
One last comment:  many modern PDKs provide SPICE-compatible versions, but they are often designed explicitly to be used with HSPICE, and models they have buried in these often-complex libraries frequently assume that "M" can be used on any device.  Addition of multiplicity to Xyce devices has been precisely to enable users of these PDKs to use them in Xyce.

Compatibility with  PDKs is one of the more significant drivers for all the HSPICE compatibility stuff you'll find showing up in Xyce development these days.  Multiplicity using "M" is one of those things.

**Reply by ngwood** [reply]
Thank you for such a thorough answer. My curiosity has been satisfied and I've been given some things to think about.
